src/c++/perf_analyzer/perf_utils.cc

```diff
@@ -423,4 +425,292 @@ ScheduleDistribution<Distribution::CONSTANT>(const double request_rate)
+        "`LD_LIBRARY_PATH` environment variable path.");
+  }
+
+  int rank{0};
+
+  MPIImplementation implementation{GetMPIImplementation()};
+  if (implementation == MPIImplementation::MPICH) {
+    int (*MPI_Comm_rank)(
+        int, int*){(int (*)(int, int*))dlsym(handle, "MPI_Comm_rank")};
+    if (MPI_Comm_rank == nullptr) {
+      throw std::runtime_error(
+          "Unable to obtain address of `MPI_Comm_rank` symbol.");
+    }
+
+    MPI_Comm_rank(MPI_COMM_WORLDMPICH(), &rank);
+  } else if (implementation == MPIImplementation::OpenMPI) {
+    int (*MPI_Comm_rank)(
+        void*, int*){(int (*)(void*, int*))dlsym(handle, "MPI_Comm_rank")};
+    if (MPI_Comm_rank == nullptr) {
+      throw std::runtime_error(
+          "Unable to obtain address of `MPI_Comm_rank` symbol.");
+    }
+
+    MPI_Comm_rank(MPI_COMM_WORLDOpenMPI(), &rank);
+  }
+
+  return rank;
+}
+
+void
+MPI_BcastIntWorld(void* buffer, int count, int root)
+{
+  void* handle{GetMPIHandle()};
+  if (handle == nullptr) {
+    throw std::runtime_error(
+        "Unable to load MPI library. Check that 'libmpi.so' is on "
+        "`LD_LIBRARY_PATH` environment variable path.");
+  }
+
+  MPIImplementation implementation{GetMPIImplementation()};
+  if (implementation == MPIImplementation::MPICH) {
+    int (*MPI_Bcast)(void*, int, int, int, int){
+        (int (*)(void*, int, int, int, int))dlsym(handle, "MPI_Bcast")};
+    if (MPI_Bcast == nullptr) {
+      throw std::runtime_error(
+          "Unable to obtain address of `MPI_Bcast` symbol.");
+    }
+
+    MPI_Bcast(buffer, count, MPI_INTMPICH(), root, MPI_COMM_WORLDMPICH());
+  } else if (implementation == MPIImplementation::OpenMPI) {
+    int (*MPI_Bcast)(void*, int, void*, int, void*){
+        (int (*)(void*, int, void*, int, void*))dlsym(handle, "MPI_Bcast")};
+    if (MPI_Bcast == nullptr) {
+      throw std::runtime_error(
+          "Unable to obtain address of `MPI_Bcast` symbol.");
+    }
+
+    MPI_Bcast(buffer, count, MPI_INTOpenMPI(), root, MPI_COMM_WORLDOpenMPI());
+  }
+}
+
+void
+MPI_Finalize()
+{
+  void* handle{GetMPIHandle()};
+  if (handle == nullptr) {
+    return;
+  }
+
+  int (*MPI_Finalize)(){(int (*)())dlsym(handle, "MPI_Finalize")};
+  if (MPI_Finalize == nullptr) {
+    throw std::runtime_error(
+        "Unable to obtain address of `MPI_Finalize` symbol.");
+  }
+
+  MPI_Finalize();
+}
+
+void*
+MPI_COMM_WORLDOpenMPI()
+{
+  void* handle{GetMPIHandle()};
+  if (handle == nullptr) {
+    throw std::runtime_error(
+        "Unable to load MPI library. Check that 'libmpi.so' is on "
+        "`LD_LIBRARY_PATH` environment variable path.");
+  }
+  void* MPI_COMM_WORLD{dlsym(handle, "ompi_mpi_comm_world")};
+  if (MPI_COMM_WORLD == nullptr) {
+    throw std::runtime_error(
+        "Unable to obtain address of `ompi_mpi_comm_world` symbol.");
+  }
+  return MPI_COMM_WORLD;
+}
+
+int
+MPI_COMM_WORLDMPICH()
+{
+  int MPI_COMM_WORLD{0x44000000};
```

Comment:
Shouldn't these be defined as constants? May be MACROS?

Reply:
Defined as constant 👍